unifai/memory/chroma.py

```diff
@@ -0,0 +1,274 @@
+import asyncio
+from typing import List, Optional, Dict, Any, Union, Sequence, cast, TypeVar, Mapping
+from uuid import UUID
+import chromadb
+from chromadb.config import Settings
+from chromadb.api import Collection
+from chromadb.types import Where, WhereDocument
+import numpy as np
+from numpy.typing import NDArray
+
+from .base import Memory, ChromaConfig, StorageType
+from .exceptions import EmptyContentError, CollectionError, ConnectionError, MemoryError
+from .protocols import MemoryManager
+from .utils import serialize_memory, deserialize_memory
+
+# Type definitions
+T = TypeVar('T')
+EmbeddingType = Union[NDArray[np.float32], List[float], Sequence[float]]
+WhereType = Dict[str, Any]
+MetadataType = Dict[str, Union[str, int, float, bool]]
+ChromaGetResult = Dict[str, Any]
+ChromaQueryResult = Dict[str, Any]
+
+def safe_cast(obj: Any, cls: type[T]) -> T:
+    return cast(cls, obj)
+
+class ChromaMemoryManager(MemoryManager):
+    def __init__(self, config: ChromaConfig):
+        self.config = config
+        try:
+            if config.storage_type == StorageType.HTTP:
+                self.client = chromadb.HttpClient(
+                    host=config.host,
+                    port=config.port,
+                    ssl=False,
+                    headers=None,
+                    settings=Settings(),
+                )
+            else:  # PERSISTENT
```

Comment:
I would recommend use else if here and returns error on else (invalid storage_type).
And maybe move try catch into each if condition like